ansible_base/lib/dynamic_config/settings_logic.py

```diff
@@ -19,55 +34,93 @@ def get_dab_settings(
     oauth2_provider: Optional[dict] = None,
     caches: Optional[dict] = None,
     templates: Optional[list[dict]] = None,
-) -> dict:
+) -> dict:  # pragma: no cover
+    """This function is deprecated and will be removed in a future version.
+    Please use `ansible_base.lib.dynamic_config.factory` instead.
+    """
+    # Controller integration CI fails on warnings, so not warning here
+    # warnings.warn(
+    #     "The function `get_dab_settings` is deprecated and will be removed in a future version. "
+    #     "Please use `ansible_base.lib.dynamic_config.factory` instead.",
+    #     DeprecationWarning,
+    #     stacklevel=2,
+    # )
+    settings = {
+        "INSTALLED_APPS": installed_apps,
+        "REST_FRAMEWORK": rest_framework or {},
+        "MIDDLEWARE": middleware or [],
+    }
+    if spectacular_settings is not None:
+        settings["SPECTACULAR_SETTINGS"] = spectacular_settings
+    if authentication_backends is not None:
+        settings["AUTHENTICATION_BACKENDS"] = authentication_backends
+    if oauth2_provider is not None:
+        settings["OAUTH2_PROVIDER"] = oauth2_provider
+    if caches is not None:
+        settings["CACHES"] = caches
+    if templates is not None:
+        settings["TEMPLATES"] = templates
+    return get_mergeable_dab_settings(settings)
+
+
+def get_mergeable_dab_settings(settings: dict) -> dict:  # NOSONAR
+    """This function takes a settings dict and returns a dict of DAB settings to be merged into the Dynaconf settings.
+
+    TODO: In a future implementation this function will defer each section to specific app logic.
+    each app will have its own get_defaults function that will return the settings to be merged.
+    e.g:
+        from ansible_base.resource_registry.settings import get_defaults
+        resource_registry_settings = get_defaults(settings)
+        dab_data.update(resource_registry_settings)
+    The call here can be explicit, or dynamic based on the INSTALLED_APPS list.
```

Comment:
I want to tackle this improvement in a future PR, so we merge this earlier to unblock integration with other services, and then we refactor later.